packages/core-data/src/selectors.ts

```diff
@@ -18,6 +18,26 @@ import { STORE_NAME } from './name';
 import { getQueriedItems } from './queried-data';
 import { DEFAULT_ENTITY_KEY } from './entities';
 import { getNormalizedCommaSeparable, isRawAttribute } from './utils';
+import type { Context, User, WpTemplate } from './entity-types';
+
+type State = any;
```

Comment:
we know this `any` isn't the right type because we later access properties inside of it inside this module, implying it's at least a `Record<string, any>`.

why not create the knowable properties and let it maintain a slightly-higher type value?

This is incomplete but I think suffices for the changes in this patch without introducing knowably-wrong types.

```ts
import type { Context, User, Theme, WpTemplate } from './entity-types';

interface State {
	autosaves: Record< string | number, Array< unknown > >;
	blockPatterns: Array< unknown >;
	blockPatternCategories: Array< unknown >;
	currentGlobalStylesId: string;
	currentTheme: Theme< 'edit' >;
	currentUser: User< 'edit' >;
	embedPreviews: Record< string, { html: string } >;
	entities: EntitiesState;
	themeBaseGlobalStyles: Record< string, Object >;
	themeGlobalStyleVariations: Record< string, string >;
	undo: UndoState;
	users: UserState;
}

interface EntitiesState {
	config: EntityConfig[];
	records: Record< string, unknown >;
}

interface EntityConfig {
	name: string;
	kind: string;
}

interface UndoState extends Array< Object > {
	flattenedUndo: unknown;
	offset: number;
}

interface UserState {
	queries: Record< string, RecordKey[] >;
	byId: Record< RecordKey, User< 'edit' > >;
}
```

Reply:
Good call!